catalog/core/catalog-core-federationstrategy/src/test/java/ddf/catalog/federation/impl/FederationStrategyTest.java

```diff
@@ -85,21 +89,54 @@
 
     private static final long LONG_TIMEOUT = 100;
 
-    private static final FilterFactory FILTER_FACTORY = new FilterFactoryImpl();
-
-    private static final ExecutorService EXECUTOR = Executors.newFixedThreadPool(2);
-
     private static final Logger LOGGER =
             LoggerFactory.getLogger(FederationStrategyTest.class.getName());
 
+    private FilterFactory filterFactory;
+
+    private ExecutorService executor;
+
     @Rule
     public PowerMockRule rule = new PowerMockRule();
 
+    @Before
+    public void setup() throws Exception {
+        filterFactory = new FilterFactoryImpl();
+        refreshExecutor();
+    }
+
+    @After
+    public void finish() throws Exception {
+        killAndWaitForExecutor();
+    }
+
+    /**
+     * Note that we ALWAYS call this upon terminating a test.
+     * @throws Exception
+     */
+    private void killAndWaitForExecutor() throws Exception {
+        if (executor != null && !executor.isShutdown()) {
+            executor.shutdown();
+            executor.awaitTermination(1L, TimeUnit.MINUTES);
+        }
+    }
+
+    /**
+     * Useful for test prep and cleaning the executor between key assertions within tests.
+     * @throws Exception
+     */
+    private void refreshExecutor() throws Exception {
+        killAndWaitForExecutor();
+        executor = Executors.newFixedThreadPool(2);
+    }
+
     /**
      * Tests that the framework properly times out using the default federation strategy.
      */
     @Test
-    public void testQueryTimeout() {
+    //TODO: Determine what is going wrong within the catalog to cause these exceptions
```

Comment:
Remove TODO and try to reproduce the problem. If there actually is an issue, open a new ticket.